Source/DafnyCore/Backends/GoLang/GoCodeGenerator.cs

```diff
@@ -167,21 +186,51 @@ private struct Import {
         return wr;
       }
 
+      var goModuleName = GoModuleMode ? GoModuleName + "/" : "";
+      if (moduleName.Equals("_System")) {
+        if (Options.IncludeRuntime) {
+          goModuleName = GoModuleMode ? GoModuleName + "/" : "";
+        } else {
+          goModuleName = GoModuleMode ? DafnyRuntimeGoModule : "";
+        }
+      }
       ModuleName = PublicModuleIdProtect(moduleName);
       var import = CreateImport(ModuleName, isDefault, externModule, libraryName);
-
       var filename = string.Format("{0}/{0}.go", import.Path);
       var w = wr.NewFile(filename);
       EmitModuleHeader(w);
 
+      import.Path = goModuleName + import.Path;
       AddImport(import);
 
       return w;
     }
 
-    protected override void DependOnModule(string moduleName, bool isDefault, ModuleDefinition externModule,
+    protected override void DependOnModule(Program program, ModuleDefinition module, ModuleDefinition externModule,
       string libraryName) {
-      var import = CreateImport(moduleName, isDefault, externModule, libraryName);
+      var goModuleName = "";
+      if (GoModuleMode && program.Compilation.AlreadyCompiledRoots.Contains(module.Tok.Uri)) {
```

Comment:
This was required because `_System` module has a special handling. Although it gets translated from a Dafny module, it is still part of the Dafny Runtime lib so has no associated go module name. So this check ensures that we search for `go-module-name` only for the `AlreadyCompiledModules` (which comes from `--lib` flag and `_System` doesn't).

Further below, we handle the `_System` module. But based on our discussion and Dafny Team's assertion that only `_System` behaves this way, I've merged the two checks.